infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/fsm/NegotiatingState.java

```diff
@@ -73,7 +77,14 @@ public LogReplicationRuntimeState processEvent(LogReplicationRuntimeEvent event)
                 return null;
             case NEGOTIATION_COMPLETE:
                 log.info("Negotiation complete, result={}", event.getNegotiationResult());
-                ((ReplicatingState)fsm.getStates().get(LogReplicationRuntimeStateType.REPLICATING)).setReplicationEvent(event.getNegotiationResult());
+                if(tableManagerPlugin.isUpgraded()) {
+                    // Force a snapshot sync if an upgrade has been identified. This will guarantee that
```

Comment:
let us add an info log explaining that a force snapshot sync will be done as the 2 LRs have now been upgraded.

Reply:
Added, thanks!